crates/rethnet_evm/benches/state.rs

```diff
@@ -45,11 +45,106 @@ fn bench_code_by_hash(c: &mut Criterion) {
     );
 }
 
+fn bench_database_commit(c: &mut Criterion) {
+    use hashbrown::HashMap;
+    use revm::primitives::KECCAK_EMPTY;
+
+    use rethnet_eth::remote::optional_u64_from_hex;
+    use rethnet_evm::{Account, AccountInfo, StorageSlot};
+
+    #[derive(Debug, serde::Deserialize)]
+    #[serde(rename_all = "camelCase")]
+    struct AccountState {
+        #[serde(deserialize_with = "optional_u64_from_hex")]
+        nonce: Option<u64>,
+        balance: Option<U256>,
+        storage: HashMap<U256, Option<U256>>,
+        code: Option<Bytes>,
+        storage_cleared: bool,
+    }
+
+    let mut accounts_to_commit: HashMap<Address, Account> = HashMap::new();
+    let json_accounts: HashMap<Address, AccountState> = serde_json::from_str(
+        &std::fs::read_to_string(
+            "benches/fixtures/accounts_changed_in_mainnet_block_17295357.json",
+        )
+        .unwrap(),
+        /* a fresh set of account updates can be retrieved via, eg:
+            BLOCK=17295357 \
+                HARDHAT_EXPERIMENTAL_VM_MODE=ethereumjs \
+                HARDHAT_RUN_FULL_BLOCK_DUMP_STATE_TO_FILE=../../crates/rethnet_evm/benches/fixtures/accounts_changed_in_mainnet_block_${BLOCK}.json \
+                sh -c 'yarn ts-node scripts/test-run-forked-block.ts $ALCHEMY_URL $BLOCK'
+            note that this should be done from the packages/hardhat-core directory.
+        */
+    )
+    .unwrap();
+    for (address, account_state) in json_accounts.iter() {
+        let mut storage: HashMap<U256, StorageSlot> = HashMap::new();
+        for (location, value) in account_state.storage.clone() {
+            storage.insert(
+                location,
+                StorageSlot {
+                    original_value: U256::ZERO, // TODO: something better?
+                    present_value: value.unwrap_or(U256::ZERO),
+                },
+            );
+        }
+        let code = account_state
+            .code
+            .clone()
+            .map(|bytes| Bytecode::new_raw(bytes));
+        accounts_to_commit.insert(
+            *address,
+            Account {
+                info: AccountInfo {
+                    balance: account_state.balance.unwrap(),
+                    nonce: account_state.nonce.unwrap(),
+                    code: code.clone(),
+                    code_hash: code.map_or(KECCAK_EMPTY, |code| code.hash()),
+                },
+                storage,
+                storage_cleared: account_state.storage_cleared,
+                is_destroyed: false,
+                is_touched: false,
+                is_not_existing: false,
+            },
+        );
+    }
+
+    bench_sync_state_method(
+        c,
+        "DatabaseCommit:commit",
+        state_prep_no_op,
+        |state, _number_of_accounts, _, _| {
+            state.commit(accounts_to_commit.clone());
+
+            debug_assert!(json_accounts.iter().all(|(address, json)| {
```

Comment:
Use `.for_each` if you want to execute something for each element in the iterator

Reply:
I did it this way because i wanted this whole statement to be optimized out if the build doesn't include `debug_assert`s, and i was suspicious of choosing a profile (`dev`, `test`, etc), because i didn't want to depend on any particular profile continuing to have `debug_assert`s enabled, but i just discovered that i can do `#[cfg(debug_assert)]`, so i did that and used `for_each` in https://github.com/NomicFoundation/hardhat/pull/4037/commits/49ed70c5d167fa20ff47dc41a7377c250136d242